src/coreclr/jit/lower.cpp

```diff
@@ -7786,6 +7784,31 @@ void Lowering::TransformUnusedIndirection(GenTreeIndir* ind, Compiler* comp, Bas
     }
 }
 
+//------------------------------------------------------------------------
+// LowerIntrinsic: a common logic to lower INTRINSIC.
+//
+// Arguments:
+//    node - the INTRINSIC node we are lowering.
+//
+void Lowering::LowerIntrinsic(GenTreeIntrinsic* node)
+{
+    if (node->AsIntrinsic()->gtIntrinsicName == NI_System_Runtime_CompilerServices_RuntimeHelpers_IsKnownConstant)
```

Comment:
This `node->AsIntrinsic()` call is redundant as the only caller (thus far, just introduced) already called that on line 587. Is it worthwhile removing it at 587, or asserting it here or just remove the call here?

Reply:
Thanks! fixed